docs/development/local_setup.md

```diff
@@ -68,9 +70,63 @@ Currently there are no binary builds available, but it is fairly simple to build
 You can verify the installation by running following command:
 
     ```console
-    $ etcdbrctl -v
-    INFO[0000] etcd-backup-restore Version: v0.7.0-dev
-    INFO[0000] Git SHA: 38979f0
-    INFO[0000] Go Version: go1.12
-    INFO[0000] Go OS/Arch: darwin/amd64
+    ➜ etcdbrctl -v
+    INFO[0000] etcd-backup-restore Version: v0.29.0-dev     
+    INFO[0000] Git SHA: 0247d8d4                            
+    INFO[0000] Go Version: go1.22.1                         
+    INFO[0000] Go OS/Arch: darwin/arm64
     ```
+
+## Running `etcdbrctl` as a process
+
+> [!WARNING]  
+> Ensure that you do not add your `credentialDirectory` or `credential_file.json` to your version control. Easiest way to ensure this, is to avoid using `git add .`. You could add these files/directory as `.gitignore` targets, or have the credential file/directory exist in a directory which is not tracked by `git`.
+
+### Environment variables
+
+Taking the example of S3 here, one has to:
+
+* Set the `AWS_APPLICATION_CREDENTIALS` environment variable if the credentials are to be stored in a directory, where each file in the directory holds a different credential
+* Set the `AWS_APPLICATION_CREDENTIALS_JSON` if the credentials are to be stored in `.json` file.
+
+Export the following environment variables according to your host OS:
+
+* `ETCD_CONF=example/01-etcd-config.yaml`
+* `POD_NAME=POD_0`
+* `POD_NAMESPACE=POD_NAMESPACE_0`
+* One of the following based on the authentication type:
+  * `AWS_APPLICATION_CREDENTIALS=/path/to/your/credential/file/directory/s3credentials`
+  * `AWS_APPLICATION_CREDENTIALS_JSON=/path/to/your/credential/file/credential_file.json`
+
+> For example: `export ETCD_CONF=example/01-etcd-config.yaml`, etc. on Unix based OS-es, and using the `set` command in Windows.
+
+`example/01-etcd-config.yaml` has the configuration required to set up `etcdbrctl` while running `etcd` as a process alongside `etcdbrctl`.
+`POD_NAME`, and `POD_NAMESPACE` are environment variables that `etcdbrctl` expects since it is typically run as a sidecar container alongside a [`etcd`](https://github.com/gardener/etcd-wrapper) in a pod, as an instance of the `Etcd` CR as defined in [`etcd-druid`](https://github.com/gardener/etcd-druid). These values are provided to ensure `etcdbrctl` runs.
+
+### Credential structure
```

Comment:
Can we move this section to somewhere above to the line `One of the following based on the authentication type:` .Since we are asking them to export the credential file/dir, so it would it good to tell them before that, the structure and content of those files. Ideally having it in bottom-up fashion will make it easier for the reader. WDYT?

Reply:
The document first explains a higher level overview before getting into the specifics, so I think it is fine in this particular case. The section immediately after the `Environment variables` is `Credential structure` anyway.

Reply:
that is fine as well